lib/data_stores/mysql_advanced_data_store.py

```diff
@@ -26,23 +26,48 @@ class Error(data_store.Error):
   """Base class for all exceptions in this module."""
 # pylint: enable=nonstandard-exception
 
+class SafeQueue(Queue.Queue):
+  def __init__(self, maxsize=0):
```

Comment:
I should have read more closely.  Is this what you're trying to do?

```
Queue.Queue.__init__(self, maxsize=maxsize)
self.mutex = threading.RLock()
```
If so, lets do that, copy pasting a whole method is almost never a good idea.